Introducing semver + "check for updates"

mieciu opened the pull request:
This PR adds simple "check for updates" functionality. 

We take the latest release from GitHub release API and compare it to currently running version. If new version is available, we log a warning and display beautiful yellow banner in the UI:

<img width="700" alt="image" src="https://github.com/user-attachments/assets/3b637cef-1590-491f-8a48-5a2e72fb8895">

(keeping my FE skills sharp 💪)

avelanarius commented on quesma/buildinfo/build.go:
```diff
@@ -2,6 +2,73 @@
+func CheckForTheLatestVersion() (updateAvailable bool, messageBanner string) {
+	latestRelease, err := getLatestRelease()
+	if err != nil {
+		return updateAvailable, fmt.Sprintf("Failed obtaining latest Quesma version form GitHub: %v", err)
```
```suggestion
		return updateAvailable, fmt.Sprintf("Failed obtaining the latest Quesma version from GitHub: %v", err)
```

mieciu commented on quesma/main.go:
```diff
@@ -59,6 +59,9 @@ func main() {
 	}, sig, doneCh, asyncQueryTraceLogger)
 	defer logger.StdLogFile.Close()
 	defer logger.ErrLogFile.Close()
+	if upgradeAvailable, message := buildinfo.CheckForTheLatestVersion(); upgradeAvailable {
```
I might actually put that into separate gorountine

avelanarius commented on quesma/buildinfo/build.go:
```diff
@@ -2,6 +2,73 @@
+func CheckForTheLatestVersion() (updateAvailable bool, messageBanner string) {
+	latestRelease, err := getLatestRelease()
+	if err != nil {
+		return updateAvailable, fmt.Sprintf("Failed obtaining latest Quesma version from GitHub: %v", err)
```
```suggestion
		return false, fmt.Sprintf("Failed obtaining latest Quesma version from GitHub: %v", err)
```
and similarly below?

mieciu replied:
https://github.com/QuesmaOrg/quesma/pull/543/commits/c1a5112ddef5f8e28f777fe296869152d5871d41

mieciu commented:
Added small tweak in https://github.com/QuesmaOrg/quesma/pull/543/commits/9f08e31dda45bcc580d6615d0be47620479513c9 - we're still blocking on a call to GitHub, yet I have introduced a timeout so that the whole UI won't hang on GitHub outage
